Tests/SwiftDocCTests/Model/SemaToRenderNodeTests.swift

```diff
@@ -1922,6 +1922,22 @@ Document
             XCTAssertEqual(renderNode.metadata.platforms?.first?.isBeta, false)
         }
         
+        // Symbol with an empty set of availbility items.
+        
+        do {
+            
+            let (bundle, context, _) = try makeTestBundle(currentPlatforms: [
+                "Custom Name": PlatformVersion(VersionTriplet(100, 0, 0), beta: true)
+            ])
+            let reference = ResolvedTopicReference(bundleIdentifier: bundle.identifier, path: "/documentation/MyKit/MyClass", sourceLanguage: .swift)
+            let node = try context.entity(with: reference)
+            (node.semantic as? Symbol)?.availability = SymbolGraph.Symbol.Availability(availability: [])
```

Comment:
This is not a blocking comment but something to keep in mind. 

Modifying a page after the context has fully registered the data provider assumes that there are no derivatives of the original value elsewhere that was created while the context registered the data provider. 

This means that the test _could_ be exercising a different behavior than a real build would and _could_ pass even though a real build would have a bug.

I _think_ that there are no derivates of the availability information today, but there could be in the future in which case this test _could_ hide a bug because it passes and there's no other test that exercises the real-build behavior. 

Because of this, it's preferred to modify the inputs before they're registered with the context.

Reply:
It's true that data could be modified after the fact, but although this generally shouldn't be done, having this precondition adds an extra layer of security to the code.